common/rdr/Exception.h

```diff
@@ -40,12 +40,17 @@ namespace rdr {
   struct SystemException : public Exception {
     int err;
     SystemException(const char* s, int err_);
-  }; 
+  };
+
+  struct NetworkException : public Exception {
```

Comment:
"Network" is a bit generic here as the class only does the correct thing for `getaddrinfo()` errors and cannot be used for any other network problems.

Given the prefix on `gai_strerror()`, perhaps `GAIException`?